### ISSUE TITLE
Beta Release

### ISSUE THREAD

**Issue by Und3rf10w** [Und3rf10w/external_c2_framework#23]
Code in the `master` branch is considered tested and stable. Once all of the changes that need to be done for the beta release are complete, then we can push them into the `master` branch.

## Tasks
The following tasks must be completed before the codebase can be considered ready for beta release:
- [ ] Create pull request to implement all new features from the `beta_stable` branch to the `master` branch.
- [ ] Merge `beta_stable` into `master`. (Blocked by all tasks below)

### Server
- [ ] Fix #14 - implement support to handle multiple clients
- [ ] Create and fix an issue to refactor and abstract the server
- [ ] **NOTSURE**: Implement the ability to support different transports across different clients

### Client
- [x] Merge #21 - merge full python `cobalt_strike` clientcore
- [ ] Fix #22 - refactor `cobalt_strike` c2 client

### Transport
- [ ] Fix #13 - support both batch encoding
- [ ] Standardize on having the transport handle calling the encoder
- [ ] Convert all existing transports into something that can be interfaced by a common object

### Base Logic
- [ ] **NOTSURE**: Refactor a the `framework` `server` interface so that we can support the management of multiple frameworks with a `framework_manager`. This may be out of scope for the beta release.

### Documentation
- [x] Fix #19 - Create builder documentation
- [x] Create documentation for the builder

**Comment by Und3rf10w** [comment]
Pushing anticipated release date to end of May 2018. Life happens unfortunately 😞

**Comment by Und3rf10w** [comment]
I'm gonna go ahead and close this as a `wontfix` for now. Please reference #26 for additional details.

**Comment by Und3rf10w** [comment]
Reopening. I'm gonna continue with the existing codebase for now and reconsider a rewrite for a later date.

**Comment by Und3rf10w** [comment]
Should probably consider whether or not we want to move towards a saner, more standardized style, [as described here](https://github.com/Und3rf10w/external_c2_framework/issues/26#issuecomment-382254766), as a target for this release.

This SHOULDN'T be terribly difficult, especially with something like PyCharm, but would be nice to have. This could also be done after beta, but we'll see.
